src/components/AnalysisName.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { Button, Input } from 'semantic-ui-react'
+import * as app from '../store/actions/app'
+
+class AnalysisName extends React.Component {
+  static propTypes = {
+    dispatch: PropTypes.func.isRequired
+  }
+
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      isEditing: false
+    }
+
+    this.handleSubmit = this.handleSubmit.bind(this)
+    this.handleClick = this.handleClick.bind(this)
+    this.handleCancel = this.handleCancel.bind(this)
+  }
+
+  handleSubmit (event) {
+    let input = this.refs.viewName.inputRef.value
+    if (input === '') {
+      input = 'Untitled Analysis'
+    }
+    this.setState({ isEditing: false })
+    document.title = input + ' | OpenTraffic Analyst UI'
```

Comment:
If we allow for empty string inputs, this should only change the title if `input` has a non-empty value.